nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
I'm still failing to see why this needs to be exposed to the API, when the cache should be transparent to any user. I don't see a use case for it, anyone wanting tickets should be using the provided service.

> The data manager would need a way to know how to operate on the cache

What is the data manager? No external plugin should be operating on our cache, they should be using the provided API methods for the service they are interested in.